### DISCUSSION TITLE
How to replace <Esc> with ctrl-c completely?

### DISCUSSION THREAD

**Discussion by MKrbm** [neovim/neovim#24851] [Q&A] [answered]
- NVIM v0.9.1
 - Windows10

I usually prefer `<C-c>` rather than `<Esc>` or `<C-[>` which made me think that I want to replace `<Esc>` with `<C-c>` perfectly for almost all situations(command, terminal, insert, change-character).

---

The naive idea was 
```
vim.api.nvim_set_keymap('i', '<C-c>', '<Esc>', { noremap = true, silent = true })

vim.api.nvim_set_keymap('c', '<C-c>', '<Esc>', { noremap = true, silent = true })

vim.api.nvim_set_keymap('t', '<C-c>', '<Esc>', { noremap = true, silent = true })
```

But the problem is that `<C-c>` doesn't behave as expected.

For examples

 - When I use change-character mode by hitting `r` and try to leave from the mode with `<C-c>`, it replaces the character with `^c` instead of leaving.

 - When I try to leave command mode with `<C-c>`, it exits the mode with the execution of the command (When I use <Esc> instead, it will just exit from command mode)

---

Can someone help me with this?

**Comment by zeertzjq** [comment]
For the second problem, you need to remove the `c` mode mapping because the default behavior of `<C-C>` in cmd line mode is the same as a typed `<Esc>`, but different from `<Esc>` from a mapping.

The first problem is tricky. It may be a bug, but changing this behavior may break scripts that rely on the current behavior.

**Reply by MKrbm** [reply]
Yeah! You're right. I removed `c` mode mapping as you advised. Thank you.

**Comment by fynnfluegge** [comment, marked as answer]
Many vim users (same for myself) have remapped `<Esc>` to `<Caps-Lock>` in the OS settings. `System Settings… > Keyboard > Keyboard Shortcuts > Modifier Keys` for Mac. This is very convenient when using vim. Have you maybe considered this already?

**Reply by MKrbm** [reply]
No, I haven't and I actually find this keymap pretty useful! I will use this for a month and see if it fits to me.
